File: hazelcast/test/map/MultiplierPortableEntryProcessor.cpp
/*
 * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

//
// Created by İhsan Demir on 26/10/15.
//

#include "serialization/TestSerializationConstants.h"
#include "MultiplierPortableEntryProcessor.h"

#include "hazelcast/client/serialization/PortableWriter.h"
#include "hazelcast/client/serialization/PortableReader.h"

namespace hazelcast {
    namespace client {
        namespace test {
            MultiplierPortableEntryProcessor::MultiplierPortableEntryProcessor(int multiplier) : multiplier(
                    multiplier) { }

            int MultiplierPortableEntryProcessor::getFactoryId() const {
                return hazelcast::client::test::TestSerializationConstants::TEST_PORTABLE_FACTORY;
            }

            int MultiplierPortableEntryProcessor::getClassId() const {
                return hazelcast::client::test::TestSerializationConstants::TEST_PORTABLE_ENTRY_PROCESSOR;
            }

            void MultiplierPortableEntryProcessor::writePortable(serialization::PortableWriter &writer) const {
                writer.writeInt("m", multiplier);
            }

            void MultiplierPortableEntryProcessor::readPortable(serialization::PortableReader &reader) {
                multiplier = reader.readInt("m");
            }
        }
    }
}

File: java/src/main/java/MultiplierPortableEntryProcessor.java
import com.hazelcast.map.EntryBackupProcessor;
import com.hazelcast.map.EntryProcessor;
import com.hazelcast.nio.serialization.Portable;
import com.hazelcast.nio.serialization.PortableReader;
import com.hazelcast.nio.serialization.PortableWriter;

import java.io.IOException;
import java.util.Map;

/**
 * Created by ihsan on 27/10/15.
 */
public class MultiplierPortableEntryProcessor
        implements EntryProcessor<String, Integer>, Portable {
    public final int FACTORY_ID = 1;
    public final int CLASS_ID = 16;

    @Override
    public Object process(Map.Entry<String, Integer> entry) {
        return entry.getValue() * multiplier;
    }

    @Override
    public EntryBackupProcessor<String, Integer> getBackupProcessor() {
        return null;
    }

    private int multiplier;

    @Override
    public int getFactoryId() {
        return FACTORY_ID;
    }

    @Override
    public int getClassId() {
        return CLASS_ID;
    }

    @Override
    public void writePortable(PortableWriter portableWriter)
            throws IOException {
        portableWriter.writeInt("m", multiplier);
    }

    @Override
    public void readPortable(PortableReader portableReader)
            throws IOException {
        multiplier = portableReader.readInt("m");
    }
}
